packages/eas-cli/src/worker/upload.ts

```diff
@@ -109,20 +109,25 @@ export async function uploadAsync(params: UploadParams): Promise<UploadResult> {
         return retry(error);
       }
 
+      const defaultErrorMessage = `Upload of "${filePath}" failed: ${response.statusText}`;
+
       if (
         response.status === 408 ||
         response.status === 409 ||
         response.status === 429 ||
         (response.status >= 500 && response.status <= 599)
       ) {
-        const message = `Upload of "${filePath}" failed: ${response.statusText}`;
         const text = await response.text().catch(() => null);
-        return retry(new Error(text ? `${message}\n${text}` : message));
+        return retry(new Error(text ? `${defaultErrorMessage}\n${text}` : defaultErrorMessage));
+      } else if (response.status === 403) {
+        const body = await response.json();
+        const message = body?.error ?? defaultErrorMessage;
+        throw new Error(message);
```

Comment:
Should we do this on every error maybe for non-403s? I think it'd also be neat to make sure we handle failure causes of `response.json()`

Reply:
So, check if `response.json().message` exists and show it instead of the default if it does?